base_layer/core/src/transactions/transaction_components/encrypted_openings_x.rs

```diff
@@ -0,0 +1,227 @@
+                "Expected {} bytes, got {}",
+                SIZE,
+                bytes.len()
+            )));
+        }
+        let mut data_1: [u8; BORSH_64] = [0u8; BORSH_64];
+        data_1.copy_from_slice(
+            bytes
+                .get(..BORSH_64)
+                .ok_or_else(|| EncryptedOpeningsErrorX::IncorrectLength("Out of bounds 'data_1'".to_string()))?,
+        );
+        let mut data_2: [u8; BORSH_X] = [0u8; BORSH_X];
+        data_2.copy_from_slice(
+            bytes
+                .get(BORSH_64..SIZE)
+                .ok_or_else(|| EncryptedOpeningsErrorX::IncorrectLength("Out of bounds 'data_2'".to_string()))?,
+        );
+        Ok(Self { data_1, data_2 })
+    }
+
+    /// Custom convert `EncryptedOpenings` to byte vector
+    pub fn as_byte_vector(&self) -> Vec<u8> {
+        let mut bytes = Vec::with_capacity(SIZE);
+        bytes.extend_from_slice(&self.data_1);
+        bytes.extend_from_slice(&self.data_2);
+        bytes
+    }
+}
+
+impl Default for EncryptedOpeningsX {
+    fn default() -> Self {
+        Self {
+            data_1: [0u8; BORSH_64],
+            data_2: [0u8; BORSH_X],
+        }
+    }
+}
+
+/// EncryptedOpenings errors
+#[derive(Debug, Error)]
+pub enum EncryptedOpeningsErrorX {
+    #[error("Encryption failed: {0}")]
+    EncryptionFailed(Error),
+    #[error("Conversion failed: {0}")]
+    ByteArrayError(#[from] ByteArrayError),
+    #[error("Incorrect length: {0}")]
+    IncorrectLength(String),
+}
+
+// Chacha error is not StdError compatible
+impl From<Error> for EncryptedOpeningsErrorX {
+    fn from(err: Error) -> Self {
+        Self::EncryptionFailed(err)
+    }
+}
+
+impl EncryptedOpeningsX {
+    const TAG: &'static [u8] = b"TARI_AAD_VALUE_AND_MASK";
+
+    /// Encrypt the value and mask (with fixed length) using XChaCha20-Poly1305 with a secure random nonce
+    pub fn encrypt_openings(
+        encryption_key: &PrivateKey,
+        commitment: &Commitment,
+        value: MicroTari,
+        mask: &PrivateKey,
+    ) -> Result<EncryptedOpeningsX, EncryptedOpeningsErrorX> {
+        let mut openings = value.as_u64().to_le_bytes().to_vec();
+        openings.append(&mut mask.to_vec());
+        let aead_payload = Payload {
+            msg: openings.as_slice(),
+            aad: Self::TAG,
+        };
+
+        // Produce a secure random nonce
+        let mut nonce = [0u8; size_of::<XNonce>()];
+        OsRng.fill_bytes(&mut nonce);
+        let nonce_ga = XNonce::from_slice(&nonce);
+
+        let aead_key = kdf_aead(encryption_key, commitment);
+        let cipher = XChaCha20Poly1305::new(GenericArray::from_slice(aead_key.reveal()));
+        let mut ciphertext = cipher.encrypt(nonce_ga, aead_payload)?;
+        let mut ciphertext_integral_nonce = nonce.to_vec();
+        ciphertext_integral_nonce.append(&mut ciphertext);
+
+        EncryptedOpeningsX::from_bytes(ciphertext_integral_nonce.as_slice())
+    }
+
+    /// Authenticate and decrypt the value and mask
```

Comment:
Recommend a comment that this design is not key committing, so the caller should not rely solely on successful decryption to assert the expected key was used.